docs/birthday-kata.md

```diff
@@ -0,0 +1,890 @@
+unique type Body = Body Text
+```
+
+Let's also update `Employee` to have an email address
+```unison
+unique type Employee  = 
+  { id: EmployeeId
+  , name: Name
+  , birthday: Date 
+  , email: EmailAddress
+  }
+```
+
+It would be nice to add some constraints to the email address (like a regex to ensure only valid email addresses can exist) but this is pretty good so far!
+
+But maybe we can build a little bit toward the future here. The prompt told us the future will likely include sending birthday messages via other platforms, like SMS or robocalls.
+
+Maybe our model should actually be about a `Message` of which there is some relationship with an `Email`. One possible way to model this would be
+```unison
+unique type Message 
+  = Email EmailAddress Subject Body
+```
+which would give us the option of extending it in the future as needed
+```unison
+unique type Message 
+  = Email EmailAddress Subject Body
+  | SMS PhoneNumber Body
+  | TwitterDM TwitterHandle Body
+```
+
+### Summary
+There isn't one "best" way to model this, but this is one possible way:
+```unison
+unique type Employee = 
+  { id: EmployeeId
+  , name: Name
+  , birthday: Date 
+  }
+
+unique type EmployeeId = EmployeeId Text
+unique type FirstName = FirstName Text
+unique type LastName = LastName Text
+unique type Name = 
+  { first: FirstName
+  , last: LastName
+  }
+
+unique type Date = 
+  { day: Day
+  , month: Month
+  , year: Year
+  }
+unique type Month 
+  = January
+  | February
+  | March
+  | April
+  | May
+  | June
+  | July
+  | August
+  | September
+  | October
+  | November
+  | December
+unique type Day = Day Nat
+unique type Year = Year Nat
+
+unique type Message 
+  = Email EmailAddress Subject Body
+
+unique type EmailAddress = EmailAddress Text
+unique type Subject = Subject Text
+unique type Body = Body Text
+```
+
+## Abilities
+Now that we have our model, let's quickly talk about the effects/infrastructure we know we will need.
+
+One of the goals of this exercise (and generally working as a software engineer) is to decouple infrastructure from our business logic (which is often approximately the same thing as spearating pure and impure code). In OO we would use interfaces to define contracts and dependency inject in whatever implementations we need (be it postgres, aws, in memory, mocks, etc).
+
+In Unison, we leverage Abilities (also called Algebraic Effects) to accomplish something similar. See [the abilities documentation](ability-typechecking.markdown) for a more detailed overview of Abilities.
+
+In our case, let's limit ourselves to the following effects: sending a message, fetching employees from somewhere (a flatfile, a db, an api, anything), and fetching the current date. For the sake of simplicity let's ignore an exception throwing ability (and failures in general) or any other abilities that may make sense to add in a real production system.
+
+```unison
+ability MessageService where
+  send: Message -> Message
```

Comment:
While the abilities are inferred by the compiler to the right of these arrows, could we spell them out here for clarity? 
`send: Message -> {MessageService} Message` etc. 

A more general data modeling topic to ponder is if returning the `Message` is material to the `MessageService` ability, or if `send` should simply return unit `()` to represent the effect of a message having been sent.  I suspect that returning an actual message might aid in the TDD nature of the exercise, but I'd be happy to think through if there's a different way of capturing / validating these messages if you felt like the MessageService's send method is better modeled with `()`.

Reply:
Yeah I think you are right, it should probably return a `()`, or better yet `Either SendFailure ()`, though I think I might want to defer editing this to include chaining Eithers, though it would be interesting to see if Unison has a `do` notation.

I think I only wrote it this way out of laziness from an earlier implementation where I changed it to return Message so I didn't need to add an extra `tap` function (or whatever it is called when you call a function with some value and then ignore the result and return the original value).

Reply:
I haven't updated this yet, but probably will on the next pass!